SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -44,19 +44,62 @@ EntityFactory::EntityFactory(USpatialNetDriver* InNetDriver, USpatialPackageMapC
 {
 }
 
-TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActorChannel* Channel, uint32& OutBytesWritten)
+FUnrealObjectRef GetStablyNamedObjectRef(UObject* Object)
+{
+	if (Object == nullptr)
+	{
+		return FUnrealObjectRef::NULL_OBJECT_REF;
+	}
+
+	// No path in SpatialOS should contain a PIE prefix.
+	FString TempPath = Object->GetFName().ToString();
+	TempPath = UWorld::RemovePIEPrefix(TempPath);
+
+	return FUnrealObjectRef(0, 0, TempPath, GetStablyNamedObjectRef(Object->GetOuter()), true);
+}
+
+TArray<FWorkerComponentData> EntityFactory::CreateSkeletonEntityComponents(AActor* Actor)
 {
-	AActor* Actor = Channel->Actor;
 	UClass* Class = Actor->GetClass();
-	Worker_EntityId EntityId = Channel->GetEntityId();
 
-	AuthorityDelegationMap DelegationMap{};
+	TArray<FWorkerComponentData> ComponentDatas;
+	ComponentDatas.Add(Position(Coordinates::FromFVector(GetActorSpatialPosition(Actor))).CreateComponentData());
+	ComponentDatas.Add(Metadata(Class->GetName()).CreateComponentData());
+	ComponentDatas.Add(SpawnData(Actor).CreateComponentData());
+
+	if (!Class->HasAnySpatialClassFlags(SPATIALCLASS_NotPersistent))
+	{
+		ComponentDatas.Add(Persistence().CreateComponentData());
+	}
+
+	// We want to have a stably named ref if this is an Actor placed in the world.
+	// We use this to indicate if a new Actor should be created, or to link a pre-existing Actor when receiving an AddEntityOp.
+	// Previously, IsFullNameStableForNetworking was used but this was only true if bNetLoadOnClient=true.
+	// Actors with bNetLoadOnClient=false also need a StablyNamedObjectRef for linking in the case of loading from a snapshot or the server
+	// crashes and restarts.
+	TSchemaOption<FUnrealObjectRef> StablyNamedObjectRef;
+	TSchemaOption<bool> bNetStartup;
+	if (Actor->HasAnyFlags(RF_WasLoaded) || Actor->bNetStartup)
+	{
+		StablyNamedObjectRef = GetStablyNamedObjectRef(Actor);
+		bNetStartup = Actor->bNetStartup;
```

Comment:
`bNetStartup` is only set at runtime, during editing we should use `bNetLoadOnClient` (we could potentially use `bNetLoadOnClient` in both cases)

Reply:
Hmm, I think this may need even something smarter due to the comment above:
```
	// We want to have a stably named ref if this is an Actor placed in the world.
	// We use this to indicate if a new Actor should be created, or to link a pre-existing Actor when receiving an AddEntityOp.
	// Previously, IsFullNameStableForNetworking was used but this was only true if bNetLoadOnClient=true.
	// Actors with bNetLoadOnClient=false also need a StablyNamedObjectRef for linking in the case of loading from a snapshot or the server
	// crashes and restarts.
```

Reply:
Discussed offline, opted to go for a check on the world to see if it's a game or editor world to see if the actor should be stably named. Also replaced `Actor->bNetStartup` with `Actor->IsNetStartupActor()` as a variant that seems to work in the editor too.

Added a bunch more checking code to see how this behaves in the wild.